src/Operation/ListSearchIndexes.php

```diff
@@ -0,0 +1,104 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+use Countable;
+use Iterator;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Driver\Server;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnexpectedValueException;
+use MongoDB\Exception\UnsupportedException;
+use MongoDB\Model\CachingIterator;
+
+use function array_intersect_key;
+use function is_string;
+
+/**
+ * Operation for the listSearchIndexes command.
+ *
+ * @see \MongoDB\Collection::listSearchIndexes()
+ * @see https://mongodb.com/docs/manual/reference/command/listSearchIndexes/
+ */
+class ListSearchIndexes implements Executable
+{
+    /** @var string */
+    private $databaseName;
+
+    /** @var string */
+    private $collectionName;
+
+    /** @var array */
+    private $listSearchIndexesOptions;
+
+    /** @var array */
+    private $aggregateOptions;
+
+    /** @var Aggregate */
+    private $aggregate;
+
+    /**
+     * Constructs an aggregate command for listing Atlas Search indexes
+     *
+     * @param string $databaseName   Database name
+     * @param string $collectionName Collection name
+     * @param array  $options        Command options
+     */
+    public function __construct(string $databaseName, string $collectionName, array $options = [])
+    {
+        if (isset($options['name']) && ! is_string($options['name'])) {
+            throw InvalidArgumentException::invalidType('"name" option', $options['name'], 'string');
+        }
+
+        if (isset($options['name']) && $options['name'] === '') {
+            throw new InvalidArgumentException('"name" option cannot be empty');
+        }
+
+        $this->databaseName = $databaseName;
+        $this->collectionName = $collectionName;
+        $this->listSearchIndexesOptions = array_intersect_key($options, ['name' => 1]);
+        $this->aggregateOptions = array_intersect_key($options, ['batchSize' => 1, 'collation' => 1, 'comment' => 1, 'maxTimeMS' => 1, 'readConcern' => 1, 'readPreference' => 1, 'session' => 1, 'typeMap' => 1]);
+
+        $this->aggregate = $this->createAggregate();
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @return Iterator&Countable
+     * @see Executable::execute()
+     * @throws UnexpectedValueException if the command response was malformed
+     * @throws UnsupportedException if collation or read concern is used and unsupported
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function execute(Server $server): Iterator
+    {
+        $cursor = $this->aggregate->execute($server);
+
+        return new CachingIterator($cursor);
+    }
+
+    private function createAggregate(): Aggregate
+    {
+        $pipeline = [
+            ['$listSearchIndexes' => (object) $this->listSearchIndexesOptions],
```

Comment:
Not an action item but just FYI: I saw that the [design document](https://docs.google.com/document/d/1rCU7vLmMEn-87SuSZ3xnnSpnkwoWpekyOl15NPIqTJk/edit) mentioned a `listSearchIndexes` command. I asked why the spec wasn't using that in [this #dbx-devs thread](https://mongodb.slack.com/archives/C72LB5RPV/p1687803718544879]). The server command predated the aggregation stage, and will likely be removed before 7.0 is released.

This will leave us with just the pipeline stage as the API for listing search indexes.